### PR TITLE
Update vSphere CCM to v1.25.0, v1.24.2 v1.23.2 and CSI driver to v2.7.0

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

- Update vSphere CCM to v1.25.0, v1.24.2, v1.23.2
- Update vSphere CSI driver to v2.7.0

**Which issue(s) this PR fixes**:
xref #2406 

**What type of PR is this?**

/kind feature

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
- Update vSphere CCM to v1.25.0, v1.24.2, v1.23.2
- Update vSphere CSI driver to v2.7.0
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: addons/ccm-vsphere/ccm-vsphere.yaml [hunk 1]
```diff
@@ -163,30 +163,29 @@ spec:
...
       nodeSelector:
         node-role.kubernetes.io/control-plane: ""
-      securityContext:
-        runAsUser: 1001
       tolerations:
         - key: node.cloudprovider.kubernetes.io/uninitialized
           value: "true"
           effect: NoSchedule
-        - key: node-role.kubernetes.io/control-plane
+        - key: node-role.kubernetes.io/master
```
Comment (kron4eg) [inline addons/ccm-vsphere/ccm-vsphere.yaml:172]: Is this correct?

Comment (xmudrii) [reply addons/ccm-vsphere/ccm-vsphere.yaml:172]: Yes. Upstream manifest switched the order of tolerations for some reason, but that doesn't really change anything.

**Pull request merged**